python/deltalake/table.py

```diff
@@ -738,14 +757,32 @@ def z_order(
             file compaction. Defaults to number of CPUs. More concurrent tasks can make compaction
             faster, but will also use more memory.
         :param max_spill_size: the maximum number of bytes to spill to disk. Defaults to 20GB.
+        :param min_commit_interval: minimum interval in seconds or as timedeltas before a new commit is
+            created. Interval is useful for long running executions. Set to 0 or timedelta(0), if you
+            want a commit per partition.
         :return: the metrics from optimize
+
+        Examples:
+
+        Use a timedelta object to specify the seconds, minutes or hours of the interval.
+        >>> from deltalake import DeltaTable
+        >>> from datetime import timedelta
+        >>> dt = DeltaTable("tmp")
+        >>> time_delta = timedelta(minutes=10)
+        >>> dt.optimize.compact(min_commit_interval=time_delta)
         """
+        if isinstance(min_commit_interval, timedelta):
+            commit_interval = min_commit_interval.seconds
```

Comment:
I think this isn't quite right:

```python
>>> from datetime import timedelta
>>> timedelta(days=5).seconds
0
>>> timedelta(days=5).total_seconds()
432000.0
```

```suggestion
            commit_interval = min_commit_interval.total_seconds()
```

Reply:
@wjones127 Strange, for minutes and seconds it gave the correct amount. Anyway I've changed it to that method, and fixed the issue where I was using | operator (my local python is always 3.10, so it became second nature to use that always). All checks should pass now.